added release notes for 10.15.0.358 (#199)
SoftwareAG/c8y-releasenotes: content/release-10-15-0/devices-10-15-0-bundle/10-15-0-358.md

---
weight: 21
title: Release 10.15.0.358
layout: bundle
---

<!--10.15.0.330 - 10.15.0.358-->

### Improvements

<div><table ><colgroup>
<col style="width: 15%;"><col style="width: 70%;"><col style="width: 15%;"></colgroup>
<thead><tr>
<th>
Component</th>
<th>
Description</th>
<th>
Issue</th>
</tr>
</thead><tbody>

<tr>
<td>LWM2M</td>

> <td>The performance of the <code>migrateLwm2mDevices</code> operation has been improved. New command line arguments have been introduced with the operation. A list of legacy LWM2M devices can be specified directly from the shell command. Moreover, the migration of the LWM2M client registration objects can be skipped by using an argument. Refer to <a href="https://cumulocity.com/guides/protocol-integration/lwm2m/#migration-of-the-lwm2m-devices" class="no-ajaxy">LWM2M > LWM2M connector device > Migration of the LWM2M devices</a> in the <i>Protocol integration guide</i> for details.</td>

Author: ```suggestion
<td>The performance of the <code>migrateLwm2mDevices</code> operation has been improved. New command line arguments have been introduced with the operation. A list of legacy LWM2M devices can be specified directly from the shell command. Moreover, the migration of the LWM2M client registration objects can be skipped by using an argument. For details, refer to <a href="https://cumulocity.com/guides/protocol-integration/lwm2m/#migration-of-the-lwm2m-devices" class="no-ajaxy">LWM2M > LWM2M connector device > Migration of the LWM2M devices</a> in the <i>Protocol integration guide</i>.</td>
```

> <td>The performance of the <code>migrateLwm2mDevices</code> operation has been improved. New command line arguments have been introduced with the operation. A list of legacy LWM2M devices can be directly specified from the shell command. Moreover, the migration of the LWM2M client registration objects can also be skipped using an argument. Refer to <a href="https://cumulocity.com/guides/protocol-integration/lwm2m/#migration-of-the-lwm2m-devices" class="no-ajaxy">LWM2M > LWM2M connector device > Migration of the LWM2M devices</a> in the <i>Protocol integration guide</i> for more details about the usage.</td>

Reviewer: ```suggestion
<td>The performance of the <code>migrateLwm2mDevices</code> operation has been improved. New command line arguments have been introduced with the operation. A list of legacy LWM2M devices can be specified directly from the shell command. Moreover, the migration of the LWM2M client registration objects can be skipped by using an argument. Refer to <a href="https://cumulocity.com/guides/protocol-integration/lwm2m/#migration-of-the-lwm2m-devices" class="no-ajaxy">LWM2M > LWM2M connector device > Migration of the LWM2M devices</a> in the <i>Protocol integration guide</i> for details.</td>
```

<tr>
<td>Connectivity</td>
<td>Due to compatibility issues of the connectivity microservice with the Kite provider, an error occurred on the metrics page, when the monthly consumption data was an out-of-scope integer. This has been fixed by changing the data type.</td>
<td>DM-1566</td>
</tr>

<tr>
<td>Device Management</td>
<td>When removing a device which has child devices, users can now select either to remove a device user or to remove all child devices. Prior to this change, it was possible to select both options in the UI, but such an action is not supported by the backend.</td>
<td>MTM-45277</td>
</tr>

<tr>
<td>Device Management</td>
<td>In the advanced-software-mgmt microservice creating or deleting a software package with special characters caused an error. This has been fixed by escaping them.</td>
<td>DM-2026</td>
</tr>

<tr>
<td>Device Management</td>
<td>SmartREST Inventory GET templates created in the UI did not generate responses when there was no external ID type declared in the template. This issue has been addressed for both existing and newly created templates.</td>
<td>DM-2126</td>
</tr>

<tr>
<td>LWM2M</td>
<td>Two new LWM2M shell commands have been added.

> <br>- The new <code>executelegacy</code> command allows LWM2M execute requests with non-standard LWM2M parameters. The behavior of this operation resembles the semantics of the existing <code>execute</code> operation until version 10.15.

Author: This is not meant. It is really "execute requests" as noun.

> <br>- The new <code>coap</code> shell command enables making raw CoAP requests to devices to facilitate non-standard communication in exceptional cases.

Author: ```suggestion
<br>- The new <code>coap</code> shell command allows raw CoAP requests to devices to facilitate non-standard communication in exceptional cases.
```

> <br>- The new <code>coap</code> shell command enables making raw CoAP requests to devices to facilitate non-standard communication in exceptional cases.

Reviewer: who is enabled? the user or LWM2M?

Author: Does that matter? I would assume both.

Reviewer: i'm asking because i'd use an object with "enable"

Reviewer: i like :)
